### PR TITLE
update eventing-crd documentation

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- enhance descriptions in the eventin crds
- update eventing-crd documentation

**Related issue(s)**

### CODE REVIEW

**Review by a-thaler — APPROVED** [review]

File: components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml [hunk 1]
```diff
@@ -68,16 +68,19 @@ spec:
                   required only for BEB
                 type: string
               conditions:
-                description: Conditions defines the status of the Controller and the
-                  EPP
+                description: Defines the status of the Controller and the EPP
                 items:
                   properties:
                     lastTransitionTime:
+                      description: Defines when the condition status was changes last
```
Comment (pbochynski) [inline components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml:75]:
```suggestion
                      description: Defines when the condition status was last changed
```

**Review by pbochynski — APPROVED** [review]

**Review by pbochynski — DISMISSED** [review]

File: components/eventing-controller/api/v1alpha2/subscription_types.go [hunk 2]
```diff
@@ -18,51 +18,54 @@ type TypeMatching string
...
 	ID string `json:"id,omitempty"`
 
-	// Sink defines endpoint of the subscriber
+	// Kubernetes Service that should be used as a target for the events that match the Subscription.
+	// Must exist in the same namespace as the Subscription.
 	Sink string `json:"sink"`
 
-	// TypeMatching defines the type of matching to be done for the event types.
+	// Defines how types should be handled.<br />
+	// - `standard`: backend specific logic will be applied to the configured source and types.<br />
```
Comment (mmitoraj) [inline components/eventing-controller/api/v1alpha2/subscription_types.go:32]:
```suggestion
	// - `standard`: backend-specific logic will be applied to the configured source and types.<br />
```

File: components/eventing-controller/api/v1alpha2/subscription_types.go [hunk 3]
```diff
@@ -18,51 +18,54 @@ type TypeMatching string
...
-	// Ready defines the overall readiness status of a Subscription.
+	// Overall readiness of the Subscription.
 	Ready bool `json:"ready"`
 
-	// Types defines the filter's event types after cleanup for use with the configured backend.
+	// List of event types after cleanup for use with the configured backend.
 	Types []EventType `json:"types"`
 
-	// Backend contains backend specific status which are only applicable to the active backend.
+	// Backend-specific status which are only applicable to the active backend.
```
Comment (mmitoraj) [inline components/eventing-controller/api/v1alpha2/subscription_types.go:60]:
```suggestion
	// Backend-specific status which is only applicable to the active backend.
```

File: components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml [hunk 4]
```diff
@@ -67,16 +67,19 @@ spec:
                   required only for BEB
                 type: string
               conditions:
-                description: Conditions defines the status of the Controller and the
-                  EPP
+                description: Defines the status of the Controller and the EPP
```
Comment (mmitoraj) [inline components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml:70]:
```suggestion
                description: Defines the status of the Controller and the EPP
```
```suggestion
                description: Defines the status of the Controller and the EPP.
```

File: components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml [hunk 5]
```diff
@@ -67,16 +67,19 @@ spec:
...
                 items:
                   properties:
                     lastTransitionTime:
+                      description: Defines when the condition status was changes last
```
Comment (mmitoraj) [inline components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml:74]:
```suggestion
                      description: Defines the date of the last condition status change.
```

File: components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml [hunk 6]
```diff
@@ -67,16 +67,19 @@ spec:
...
                       format: date-time
                       type: string
                     message:
+                      description: Provides more detail about the condition status
```
Comment (mmitoraj) [inline components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml:78]:
```suggestion
                      description: Provides more details about the condition status.
```

File: components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml [hunk 7]
```diff
@@ -67,16 +67,19 @@ spec:
...
+                        change
                       type: string
                     reason:
+                      description: Defines why the condition status changed
```
Comment (mmitoraj) [inline components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml:82]:
```suggestion
                      description: Defines the reason for the condition status change.
```

**Review by mmitoraj — COMMENTED** [review]
Great effort and a nice job! I hope I caught everything but if not, please bear with me once I have another look. Thank you!

File: components/eventing-controller/api/v1alpha1/subscription_types.go [hunk 8]
```diff
@@ -14,42 +14,44 @@ var Finalizer = GroupVersion.Group
...
-	// GrantType defines grant type for OAuth2
+	// Defines grant type for OAuth2
 	GrantType string `json:"grantType"`
 
-	// ClientID defines clientID for OAuth2
+	// Defines clientID for OAuth2
 	ClientID string `json:"clientId"`
 
-	// ClientSecret defines client secret for OAuth2
+	// Defines client secret for OAuth2
```
[the remark this answers is not in the document]
Comment (k15r) [reply components/eventing-controller/api/v1alpha1/subscription_types.go:27]: why `Client Secret` and not `client secret`?

File: components/eventing-controller/api/v1alpha1/subscription_types.go [hunk 9]
```diff
@@ -14,42 +14,44 @@ var Finalizer = GroupVersion.Group
...
+	// Defines scope for OAuth2
 	Scope []string `json:"scope,omitempty"`
 }
 
 // ProtocolSettings defines the CE protocol setting specification implementation.
 // TODO: Remove it when depreciating code of v1alpha1
 type ProtocolSettings struct {
-	// ContentMode defines content mode for eventing based on BEB.
+	// Defines the content mode for eventing based on BEB.
+	// Must be one of "BINARY", "STRUCTURED".
```
[the remark this answers is not in the document]
Comment (k15r) [reply components/eventing-controller/api/v1alpha1/subscription_types.go:41]: it must be all caps, but yes, either BINARY or STRUCTURED

File: components/eventing-controller/api/v1alpha1/subscription_types.go [hunk 10]
```diff
@@ -14,42 +14,44 @@ var Finalizer = GroupVersion.Group
...
+	// Default:
```
[the remark this answers is not in the document]
Comment (k15r) [reply components/eventing-controller/api/v1alpha1/subscription_types.go:42]: oops, no

File: components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml [hunk 11]
```diff
@@ -52,41 +52,48 @@ spec:
             description: EventingBackendStatus defines the observed state of EventingBackend.
             properties:
               backendType:
-                description: Specifies the backend type used. Allowed values are "BEB"
-                  and "NATS"
+                description: Specifies the backend type used. The value is either
+                  BEB, or NATS.
```
Comment (mmitoraj) [inline components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml:56]:
```suggestion
                  `BEB`, or `NATS`.
```

File: components/eventing-controller/config/crd/bases/eventing.kyma-project.io_subscriptions.yaml [hunk 12]
```diff
@@ -106,50 +110,49 @@ spec:
                 - filters
                 type: object
               id:
-                description: ID is the unique identifier of Subscription, read-only.
+                description: Unique identifier of Subscription, read-only.
```
Comment (mmitoraj) [inline components/eventing-controller/config/crd/bases/eventing.kyma-project.io_subscriptions.yaml:113]:
```suggestion
                description: Unique identifier of the Subscription, read-only.
```

File: components/eventing-controller/config/crd/bases/eventing.kyma-project.io_subscriptions.yaml [hunk 13]
```diff
@@ -159,95 +162,100 @@ spec:
                     type: object
                 type: object
               sink:
-                description: Sink defines endpoint of the subscriber
+                description: Kubernetes Service that should be used as a target for
+                  the events that match the Subscription. Must exist in the same namespace
```
Comment (mmitoraj) [inline components/eventing-controller/config/crd/bases/eventing.kyma-project.io_subscriptions.yaml:166]:
```suggestion
                  the events that match the Subscription. Must exist in the same Namespace
```

File: docs/05-technical-reference/00-custom-resources/evnt-01-subscription.md [hunk 14]
```diff
@@ -35,15 +35,116 @@ spec:
...
+| **conditions.&#x200b;status** (required) | string | Status of the condition. The value is either `True`, `False`, or `Unknown`. |
+| **conditions.&#x200b;type**  | string | Short description of the condition. |
+| **ready** (required) | boolean | Overall readiness of the Subscription. |
+| **types** (required) | \[\]object | List of event types after cleanup for use with the configured backend. |
+| **types.&#x200b;cleanType** (required) | string | Event type after it was cleaned up from backend compatible characters. |
+| **types.&#x200b;originalType** (required) | string | Event type as specified in the Subscription spec. |
+
+### Subscription.eventing.kyma-project.io/v1alpha1
+
+>**CAUTION**: v1alpha1 is deprecated as of Kyma 2.14.X
```
Comment (mmitoraj) [inline docs/05-technical-reference/00-custom-resources/evnt-01-subscription.md:87]:
```suggestion
>**CAUTION**: The v1alpha1 API version is deprecated as of Kyma 2.14.X.
```

File: components/eventing-controller/api/v1alpha1/subscription_types.go [hunk 15]
```diff
@@ -88,6 +89,8 @@ func (bf *BEBFilter) hash() (uint64, error) {
 
 // BEBFilters defines the list of BEB filters.
 type BEBFilters struct {
+	// Contains a URI-reference to CloudEvent filter dialect. See
```
Comment (mmitoraj) [inline components/eventing-controller/api/v1alpha1/subscription_types.go:92]:
```suggestion
	// Contains a `URI-reference` to the CloudEvent filter dialect. See
```

**Review by mmitoraj — COMMENTED** [review]
Just a few final comments. And I'm not sure why the Spec table doesn't show in the preview?

Screenshot 2023-05-17 at 09 06 46

File: components/eventing-controller/api/v1alpha1/subscription_types.go [hunk 16]
```diff
@@ -136,101 +140,100 @@ func MergeSubsConfigs(config *SubscriptionConfig, defaults *env.DefaultSubscript
...
+	// Defines the configurations that have been applied to the eventing backend when creating this Subscription.
 	// +optional
 	Config *SubscriptionConfig `json:"config,omitempty"`
 }
 
 // Subscription is the Schema for the subscriptions API.
 // +kubebuilder:object:root=true
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 // +kubebuilder:subresource:status
+// +kubebuilder:deprecatedversion:warning=v1alpha1 is deprecated as of Kyma 2.14.X
```
Comment (marcobebway) [inline components/eventing-controller/api/v1alpha1/subscription_types.go:233]: 👍

File: components/eventing-controller/api/v1alpha2/subscription_types.go [hunk 17]
```diff
@@ -18,51 +18,54 @@ type TypeMatching string
 
 var Finalizer = GroupVersion.Group
 
-// SubscriptionSpec defines the desired state of Subscription.
+// Defines the desired state of the Subscription.
```
Comment (marcobebway) [inline components/eventing-controller/api/v1alpha2/subscription_types.go:21]: Comment should start with the name of the struct `SubscriptionSpec`.

File: hack/table-gen/Makefile [hunk 18]
```diff
@@ -1,5 +1,5 @@
 .PHONY: generate
-generate: telemetry-docs cra-docs
+generate: telemetry-docs cra-docs eventing-docs
```
Comment (marcobebway) [inline hack/table-gen/Makefile:2]: Where is `telemetry-docs cra-docs eventing-docs` defined?

File: components/eventing-controller/api/v1alpha2/subscription_types.go (hunk 17 above)
Comment (marcobebway) [reply components/eventing-controller/api/v1alpha2/subscription_types.go:21]: Agreed to approve as it is for now.

**Review by marcobebway — DISMISSED** [review]

**Review by mmitoraj — APPROVED** [review]

**Review by marcobebway — APPROVED** [review]

**Review by a-thaler — APPROVED** [review]

**Pull request merged**